### DISCUSSION TITLE
Tailwind & Tailwind JIT could use in React-native project?

### DISCUSSION THREAD

**Discussion by l2D** [tailwindlabs/tailwindcss#5161] [Help]
Hello, Tailwindcss comm.

I use Tailwindcss on Vue and Nuxt, I love it!

And I have a question about how to use Tailwind & Tailwind JIT could use in a React-native project?

If you've some suggestions or tutorials, please tell me. Thank you.

**Comment by NovusOpus** [comment]
So I found this tutorial: [Article Link](https://blog.logrocket.com/why-you-should-use-tailwind-css-with-react-native/) , Or you can check/clone the final [GitHub Repo](https://github.com/IdaszakDaniel/TailwindCssReactNative).

While the tutorial doesn't include a **Jit Approach**, Searching through `tailwind-react-native-classnames` (The package for using Tailwind with HTML Classes like you do in Nuxt, Next or in general Web Dev, I found this quote from the Developer in a Jit related Issue: 

> I just recently found out about Tailwind JIT, so I'm not 100%, but I think my answer would be similar to what I said in #8. JIT sounds awesome for web projects, but I'm not sure it's all that useful for RN projects -- it's a totally different paradigm. We're not shipping a .css file, we're configuring a function to return react-native style objects when passed tailwind classes. I suppose there might be an effect for RN-web, but that's not a use case that I currently think this project is optimized for, and I'm not sure it would be the right approach to try to support every web-based feature for people who are using RN for web.

The good news is though,  The same Developer posted this message yesterday:

> I've begun tentative work on a 2.x version of this library that is a complete rewrite, and functions much closer to the new JIT paradigm in tailwindcss. It no longer parses CSS ahead of time generating a style map. Instead, it will be able to resolve and parse tailwind classes on the fly, including support for arbitrary values like mx-[33px], also dark mode (#15), some quasi-media queries, and some other good stuff. It will be heavily optimized for speed, so it should perform (hopefully) even better than v1.

So a new version with Jit Support should be in the works and hopefully available soon, You can check or follow the Jit Issue/topic [here](https://github.com/jaredh159/tailwind-react-native-classnames/issues/16). 

Hopes this helps you with a place to start from, You can try to manually add Jit (I may try it later) into this approach.

**Comment by Lellands** [comment]
Tailwind can take 3–8s to initially compile using our CLI, and upwards of 30–45s in webpack projects because webpack struggles with large CSS files. This library can compile even the biggest projects in about 800ms (with incremental rebuilds as fast as 3ms), no matter what build tool you're using.

[UPSers](https://www.upsers.biz/)

**Comment by huibizhang** [comment]
No problem, I find a solution for this.

You can see this reply of issue

https://github.com/tailwindlabs/tailwindcss/issues/5132#issuecomment-894549642
